src/ralph/backends/data/clickhouse.py

```diff
@@ -0,0 +1,524 @@
+        """Instantiates the ClickHouse configuration.
+
+        Args:
+            settings (ClickHouseDataBackendSettings): The ClickHouse data backend
+                settings.
+            host (str): ClickHouse server host to connect to.
+            port (int): ClickHouse server port to connect to.
+            database (str): ClickHouse database to connect to.
+            event_table_name (str): Table where events live.
+            username (str): ClickHouse username to connect as (optional).
+            password (str): Password for the given ClickHouse username (optional).
+            client_options (dict): A dictionary of valid options for the ClickHouse
+                client connection.
+            locale_encoding (str): The locale encoding to use when none is provided.
+            _client (clickhouse_driver.Client): The ClickHouse client instance.
+
+        If username and password are None, we will try to connect as the ClickHouse
+        user "default".
+        """
+        self.host = settings.HOST
+        self.port = settings.PORT
+        self.database = settings.DATABASE
+        self.event_table_name = settings.EVENT_TABLE_NAME
+        self.username = settings.USERNAME
+        self.password = settings.PASSWORD
+        self.client_options = settings.client_options.dict()
+        self.locale_encoding = settings.LOCALE_ENCODING
+        self._client = None
+
+    @property
+    def client(self):
+        """Creates a ClickHouse client if it doesn't exist.
+
+        We do this here so that we don't interrupt initialization in the case
+        where ClickHouse is not running when Ralph starts up, which will cause
+        Ralph to hang. This client is HTTP, so not actually stateful. Ralph
+        should be able to gracefully deal with ClickHouse outages at all other
+        times.
+        """
+        if not self._client:
+            self._client = clickhouse_connect.get_client(
+                host=self.host,
+                port=self.port,
+                database=self.database,
+                username=self.username,
+                password=self.password,
+                settings=self.client_options,
+            )
+        return self._client
+
+    def status(self) -> DataBackendStatus:
+        """Checks ClickHouse connection status."""
+        try:
+            self.client.query("SELECT 1")
+        except ClickHouseError:
+            return DataBackendStatus.AWAY
+
+        return DataBackendStatus.OK
+
+    def list(
+        self, target: str = None, details: bool = False, new: bool = False
+    ) -> Iterator[Union[str, dict]]:
+        """Lists documents for a given table.
+
+        Args:
+            target (str): The table to list documents from.
+            details (bool): Get detailed archive information instead of just ids.
+            new (bool): Given the history, list only not already fetched archives.
+        """
+        sql = "SHOW TABLES"  # nosec
```

Comment:
```suggestion
        sql = "SHOW TABLES"
```
What linter error does this line raise?